url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+
+	// $ANTLR start "APRIL"
+	public final void mAPRIL() throws RecognitionException {
+		try {
+			int _type = APRIL;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:318:11: ( 'april' ( 's' )? | 'apr' ( DOT )? )
+			int alt15=2;
+			int LA15_0 = input.LA(1);
+			if ( (LA15_0=='a') ) {
+				int LA15_1 = input.LA(2);
+				if ( (LA15_1=='p') ) {
+					int LA15_2 = input.LA(3);
+					if ( (LA15_2=='r') ) {
+						int LA15_3 = input.LA(4);
+						if ( (LA15_3=='i') ) {
+							alt15=1;
+						}
+
+						else {
+							alt15=2;
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 15, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 15, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 15, 0, input);
+				throw nvae;
+			}
+
+			switch (alt15) {
+				case 1 :
+					// HiveLexer.g:318:13: 'april' ( 's' )?
+					{
+					match("april"); 
+
+					// HiveLexer.g:318:25: ( 's' )?
+					int alt13=2;
+					int LA13_0 = input.LA(1);
+					if ( (LA13_0=='s') ) {
+						alt13=1;
+					}
+					switch (alt13) {
+						case 1 :
+							// HiveLexer.g:318:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:318:33: 'apr' ( DOT )?
+					{
+					match("apr"); 
+
+					// HiveLexer.g:318:39: ( DOT )?
+					int alt14=2;
+					int LA14_0 = input.LA(1);
+					if ( (LA14_0=='.') ) {
+						alt14=1;
+					}
+					switch (alt14) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "APRIL"
+
+	// $ANTLR start "MAY"
+	public final void mMAY() throws RecognitionException {
+		try {
+			int _type = MAY;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:319:11: ( 'may' ( 's' )? )
+			// HiveLexer.g:319:13: 'may' ( 's' )?
+			{
+			match("may"); 
+
+			// HiveLexer.g:319:25: ( 's' )?
+			int alt16=2;
+			int LA16_0 = input.LA(1);
+			if ( (LA16_0=='s') ) {
+				alt16=1;
+			}
+			switch (alt16) {
+				case 1 :
+					// HiveLexer.g:319:25: 's'
+					{
+					match('s'); 
+					}
+					break;
+
+			}
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 6725:6744, 8395:8414, 8778:8797.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.